cli/src/cli_util.rs

```diff
@@ -1584,6 +1610,50 @@ Then run `jj squash` to move the resolution into the conflicted commit."#,
     }
 }
 
+#[allow(clippy::too_many_arguments)]
+fn check_reset_author_on_edit(
+    settings: &UserSettings,
+    workspace_id: &WorkspaceId,
+    // Required to build IdPrefixContext
+    user_repo: &ReadonlyUserRepo,
+    revset_extensions: &Arc<RevsetExtensions>,
+    revset_aliases_map: &RevsetAliasesMap,
+    path_converter: &RepoPathUiConverter,
+    // May be different from user_repo for MutableRepo
+    repo: &dyn Repo,
```

Comment:
I found a similar problem in `check_repo_rewritable()`. I think the "reset-author" revset belongs to the same category.
https://github.com/martinvonz/jj/pull/3948

Reply:
Yeah, I think it would be similar. Based on that, I'm thinking I could extract a `revset_helper` struct with pretty minor code changes to make it a lot cleaner.